### ISSUE TITLE
Feature Request: Convert Insight Type to Another Service Type

### ISSUE THREAD

**Issue by pedantic79** [bwp91/homebridge-platform-wemo#260]
* **Please explain your feature request in a one or two sentences.**

Feature #257 only supports outlets. Can this also be added to Wemo Insights?

* **Is your feature request related to a problem? Please describe.**

No

* **Any particular Wemo devices that this relates to?**

Wemo Insights

* **Anything else?**

This may not be possible, just would be nice because I like to see how much my air cleaner uses since it's on all the time.

**Comment by bwp91** [comment]
Hi @pedantic79 

It’s something I can look into!

The reason I didn’t initially extend this feature to Insight devices is because of the extra power characteristics that can be seen in the Eve app (wattage, usage, outlet-in-use etc). These are normally added to an outlet accessory type, i’ll need to check to see if they can be added to a purifier accessory type.

If they can, great, if they can’t, then exposing as an air purifier will still be possible but the extra features will be lost, it would basically just be “on” and “off”.

**Comment by bwp91** [comment]
Could you try updating to the latest beta version of the plugin?

And use the new 'Show As' config option inside 'Wemo Insights'?

I'm not sure if you use the Eve app at all for HomeKit? I'm unable to test this since I don't have an Insight. Feedback would be great!

**Comment by pedantic79** [comment]
I don't use Eve. Looks like it now shows up as an "Air Purifier". Only issue is that it seems to get stuck in "Turning on..."

![image](https://user-images.githubusercontent.com/11181103/123474108-1a33bc00-d5bf-11eb-966c-e473d7a59cf0.png)

**Comment by pedantic79** [comment]
More info:
When I turn off, it goes to off.
When I turn it on, it goes to on for a second or two, then switches to "Turning on..."

```
[6/25/2021, 2:11:56 PM] [Wemo] [Air Cleaner] today ontime [00:04:39] consumption [0.051 kWh] total consumption [2.110 kWh].
[6/25/2021, 2:11:57 PM] [Wemo] [Air Cleaner] current state [off].
[6/25/2021, 2:11:57 PM] [Wemo] [Air Cleaner] current consumption [0W].
[6/25/2021, 2:11:59 PM] [Wemo] [Air Cleaner] current state [off].
[6/25/2021, 2:11:59 PM] [Wemo] [Air Cleaner] current consumption [0W].
[6/25/2021, 2:12:03 PM] [Wemo] [Air Cleaner] current state [on].
[6/25/2021, 2:12:04 PM] [Wemo] [Air Cleaner] current state [on].
[6/25/2021, 2:12:07 PM] [Wemo] [Air Cleaner] current consumption [3W].
[6/25/2021, 2:12:13 PM] [Wemo] [Air Cleaner] current consumption [4W].
[6/25/2021, 2:13:07 PM] [Wemo] [Air Cleaner] today ontime [00:04:39] consumption [0.051 kWh] total consumption [2.110 kWh].
```

**Comment by pedantic79** [comment]
```
[6/25/2021, 2:17:09 PM] [Wemo] [Air Cleaner] sending update {"BinaryState":0}.
[6/25/2021, 2:17:09 PM] [Wemo] [Air Cleaner] current state [off].
[6/25/2021, 2:17:09 PM] [Wemo] [Air Cleaner] current consumption [0W].
[6/25/2021, 2:17:10 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>0|1624648557|66|279|12404|1209600|6|3730|3090572|11476993|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:10 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 0, power: 3730, todayWm: 3090572, todayOnSeconds: 279 }].
[6/25/2021, 2:17:10 PM] [Wemo] [Air Cleaner] current state [off].
[6/25/2021, 2:17:10 PM] [Wemo] [Air Cleaner] current consumption [0W].
[6/25/2021, 2:17:10 PM] [Wemo] [Air Cleaner] current consumption [4W].
[6/25/2021, 2:17:11 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<BinaryState>0|1624648630|66|279|12404|1209600|6|0|3090572|11476993</BinaryState>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:11 PM] [Wemo] [Air Cleaner] receiving update [BinaryState: 0].
[6/25/2021, 2:17:11 PM] [Wemo] [Air Cleaner] current consumption [0W].
[6/25/2021, 2:17:15 PM] [Wemo] [Air Cleaner] sending update {"BinaryState":1}.
[6/25/2021, 2:17:15 PM] [Wemo] [Air Cleaner] current state [on].
[6/25/2021, 2:17:17 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<BinaryState>8|1624648636|66|279|12404|1209600|6|2635|3090572|11476993</BinaryState>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:17 PM] [Wemo] [Air Cleaner] receiving update [BinaryState: 8].
[6/25/2021, 2:17:17 PM] [Wemo] [Air Cleaner] current state [on].
[6/25/2021, 2:17:20 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|2440|3090572|11476993|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:20 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 2440, todayWm: 3090572, todayOnSeconds: 279 }].
[6/25/2021, 2:17:20 PM] [Wemo] [Air Cleaner] current consumption [2W].
[6/25/2021, 2:17:21 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|2605|3090572|11476993|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:21 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 2605, todayWm: 3090572, todayOnSeconds: 279 }].
[6/25/2021, 2:17:21 PM] [Wemo] [Air Cleaner] current consumption [3W].
[6/25/2021, 2:17:24 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3330|3090572|11476993|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:24 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3330, todayWm: 3090572, todayOnSeconds: 279 }].
[6/25/2021, 2:17:25 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3545|3090572|11476993|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:25 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3545, todayWm: 3090572, todayOnSeconds: 279 }].
[6/25/2021, 2:17:25 PM] [Wemo] [Air Cleaner] current consumption [4W].
[6/25/2021, 2:17:28 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3700|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:28 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3700, todayWm: 3094179, todayOnSeconds: 279 }].
[6/25/2021, 2:17:28 PM] [Wemo] [Air Cleaner] today ontime [00:04:39] consumption [0.052 kWh] total consumption [2.111 kWh].
[6/25/2021, 2:17:29 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3725|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:29 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3725, todayWm: 3094179, todayOnSeconds: 279 }].
[6/25/2021, 2:17:32 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3755|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:32 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3755, todayWm: 3094179, todayOnSeconds: 279 }].
[6/25/2021, 2:17:33 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3750|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:33 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3750, todayWm: 3094179, todayOnSeconds: 279 }].
[6/25/2021, 2:17:36 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3735|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:36 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3735, todayWm: 3094179, todayOnSeconds: 279 }].
[6/25/2021, 2:17:37 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3710|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:37 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3710, todayWm: 3094179, todayOnSeconds: 279 }].
[6/25/2021, 2:17:40 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3705|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:40 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3705, todayWm: 3094179, todayOnSeconds: 279 }].
[6/25/2021, 2:17:41 PM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<InsightParams>8|1624648636|66|279|12404|1209600|6|3725|3094179|11480600|5000</InsightParams>
</e:property>
</e:propertyset>
[6/25/2021, 2:17:41 PM] [Wemo] [Air Cleaner] receiving update [InsightParams: { state: 8, power: 3725, todayWm: 3094179, todayOnSeconds: 279 }].
```

**Comment by bwp91** [comment]
Please try 3.1.1-beta.6 when it appears available

**Comment by pedantic79** [comment]
Looks good.

**Comment by bwp91** [comment]
Could you try one more beta for me? v3.1.1-beta.7.

I have tried to map the *OutletInUse* status that the insight provides to the current purifying state in HomeKit, ie

* insight OFF -> purifier OFF
* insight STANDBY -> purifier ON but IDLE
* insight ON -> purifier ON and PURIFYING

Makes sense to try to use the given info in the best way.

Also, if you ever considered having a look at the Eve HomeKit app... you should find that the current wattage and also total consumption for this insight are displayed there in the accessory information (it's something that isn't available in the Home app)

**Comment by pedantic79** [comment]
This is working even better. I checked by increasing the auto speed on the purifier to max, and the purifier shows on. Switching back moves it back to idle.

I'll take a look at Eve Homekit when I have the chance.

**Comment by bwp91** [comment]
Fab!

I believe you can change the wattage threshold via the wemo app for at what point it reports the ON as opposed to STANDBY to the plugin.

**Comment by pedantic79** [comment]
Do we have an eta for when this will be available in the stable build?

**Comment by bwp91** [comment]
within the next few days definitely. i time my releases for when i have some spare time in case of any issues!

**Comment by bwp91** [comment]
@pedantic79 

Can you update to latest beta release, v3.2.0-beta.4, and just change in your config the entry `showAsType` to `showAs`, ie

```
"showAsType": "purifier",
```

to

```
"showAs": "purifier",
```

and just confirms it still works okay?

**Comment by pedantic79** [comment]
Still working. I did notice that turning the device off it goes into "Turning off..."

```
[7/6/2021, 10:34:17 AM] [Wemo] [Air Cleaner] sending update {"BinaryState":0}.
[7/6/2021, 10:34:17 AM] [Wemo] [Air Cleaner] current state [off].
[7/6/2021, 10:34:17 AM] [Wemo] [Air Cleaner] current purifying [no].
[7/6/2021, 10:34:17 AM] [Wemo] [Air Cleaner] current consumption [0W].
[7/6/2021, 10:34:19 AM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<BinaryState>0|1625585658|74|74|12616|1209600|5|0|2257546|4448236</BinaryState>
</e:property>
</e:propertyset>
[7/6/2021, 10:34:19 AM] [Wemo] [Air Cleaner] receiving update [BinaryState: 0].
[7/6/2021, 10:34:19 AM] [Wemo] [Air Cleaner] current consumption [0W].
```

**Comment by bwp91** [comment]
Can you try 3.2.0-beta.5?

**Comment by pedantic79** [comment]
Still the same issue.

```
[7/6/2021, 11:47:21 AM] [Wemo] [Air Cleaner] sending update {"BinaryState":0}.
[7/6/2021, 11:47:22 AM] [Wemo] [Air Cleaner] current state [not purifying].
[7/6/2021, 11:47:22 AM] [Wemo] [Air Cleaner] current consumption [0W].
[7/6/2021, 11:47:23 AM] [Wemo] [uuid:Insight-1_0-231612K12002DE] incoming notification:
<e:propertyset xmlns:e="urn:schemas-upnp-org:event-1-0">
<e:property>
<BinaryState>0|1625590042|74|74|12616|1209600|5|0|2516055|4706745</BinaryState>
</e:property>
</e:propertyset>
[7/6/2021, 11:47:23 AM] [Wemo] [Air Cleaner] receiving update [BinaryState: 0].
[7/6/2021, 11:47:23 AM] [Wemo] [current state] Air Cleaner [not purifying].
```

**Comment by bwp91** [comment]
and 3.2.0-beta.6? 😛

**Comment by pedantic79** [comment]
Looks good

**Comment by bwp91** [comment]
Gonna close this now as available in v3.2 of the plugin.
Thanks!
